codecov.yml

```diff
@@ -0,0 +1,33 @@
+coverage:
+  status:
+    project:
+      default:
+        # basic
+        target: auto
+        threshold: 2% # allow cov to drop by 2% (just in case)
+    patch:
+      default:
+        threshold: 1% # allow patch
+
+ignore:
+  - "**/*.yaml"       # ignore all yaml files (Kubernetes manifests, etc...)
+  - "**/*.yml"        # same as above
+  - "**/*.md"         # ignore all markdown files, those are not relevant for building/testing
+  - "**/Dockerfile"   # ignore Dockerfiles, those are build with GH Actions
+  - "**/*.sh"         # ignore shell scripts
```

Comment:
can we maybe ignore all files that do not have any suffix? It should reduce the number of files quite a lot

Reply:
Right now, codecov only shows the files reported in the coverage files anyways, so only go files are in the report.
So I think we can go ahead with the current config file and adjust from there if we really need to.